File: app/src/main/java/com/an/trailers/di/module/AppModule.java
package com.an.trailers.di.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Application mApplication;

    public AppModule(Application mApplication) {
        this.mApplication = mApplication;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/an/trailers/di/module/AppModule.kt
package com.an.trailers.di.module

import android.app.Application
import dagger.Module
import dagger.Provides

import javax.inject.Singleton

@Module
class AppModule(private val mApplication: Application) {

//    @Provides
//    @Singleton
//    internal fun provideApplication(): Application {
//        return mApplication
//    }
}
